### ISSUE TITLE
[REF-1652] Bug when applying font_family globally

### ISSUE THREAD

**Issue by 4l3j0Ok** [reflex-dev/reflex#2392]
### Description

The 'font_family' is not applying if set in the base style (style value in rx.App kwargs). However, if set 'font-family' individually in each component, it works fine.

### To Reproduce

Try this example code on Reflex 0.3.7, after that, try on Reflex 0.3.8:

```
import reflex as rx
import pkg_resources
reflex_version = pkg_resources.get_distribution("reflex").version

# GLOBAL FONT FAMILY BUG
BASE = {
    "font_family": "Comic Sans MS"
}

@rx.page("index")
def index():
    return rx.box(
        rx.span(
            "reflex==",
            rx.span(reflex_version, color='red')
        ),
        rx.text("This is a text."),
        rx.span("This is a span.")
    )

app = rx.App(
    style = BASE
)

app.compile()
```

### Expected behavior

On 0.3.7, the font_family class should be applied, but on 0.3.8 shouldn't.

### Screenshots

| Reflex Version | Screenshot |
| -- | -- |
| reflex==0.3.7 | ![image](https://uploads.linear.app/9513bc1c-787d-499b-8c86-46952ab7013c/f2352b96-7af1-47bf-9674-f8159dfa3aa5/48ab08ea-ace9-4021-8992-9bb8f3902678?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzk1MTNiYzFjLTc4N2QtNDk5Yi04Yzg2LTQ2OTUyYWI3MDEzYy9mMjM1MmI5Ni03YWYxLTQ3YmYtOTY3NC1mODE1OWRmYTNhYTUvNDhhYjA4ZWEtYWNlOS00MDIxLTg5OTItOWJiOGYzOTAyNjc4IiwiaWF0IjoxNzI0NzIyNzk5LCJleHAiOjE3NTYyNTg3OTl9.RH_peMhkHjf7fljFdwBr5aSz1HYwqOS3tXZC2XX7ugc) |
| reflex==0.3.8 | ![image](https://uploads.linear.app/9513bc1c-787d-499b-8c86-46952ab7013c/3525ee24-d72e-44b5-af0c-5d9b6e65981c/4139b788-0c36-44ff-86b2-b3687b0b4bd2?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzk1MTNiYzFjLTc4N2QtNDk5Yi04Yzg2LTQ2OTUyYWI3MDEzYy8zNTI1ZWUyNC1kNzJlLTQ0YjUtYWYwYy01ZDliNmU2NTk4MWMvNDEzOWI3ODgtMGMzNi00NGZmLTg2YjItYjM2ODdiMGI0YmQyIiwiaWF0IjoxNzI0NzIyNzk5LCJleHAiOjE3NTYyNTg3OTl9.FZtER_uXUFSMH8_R8Hx_yCVs-e1A9GuMkpMVrjfIeME) |

### Specifics:

* Python Version: 3.11.7
* Reflex Version: 0.3.8
* OS: Ubuntu (WSL)

<sub>From [SyncLinear.com](https://synclinear.com) | [REF-1652](https://linear.app/reflex-dev/issue/REF-1652/bug-when-applying-font-family-globally)</sub>

**Comment by mouredev** [comment]
It seems that the same problem exists with other attributes such as "font_weight" or "background_color".

**Comment by picklelo** [comment]
This should be fixed in 0.3.9

**Comment by 4l3j0Ok** [comment]
Fixed in 0.3.9, as shown in the screenshot.
![image](https://github.com/reflex-dev/reflex/assets/69774461/9f57c635-14de-497f-966a-65b984e05127)
